src/main/java/com/puppycrawl/tools/checkstyle/gui/TreeTable.java

```diff
@@ -143,6 +154,33 @@ private void makeCodeSelection() {
         new CodeSelector(tree.getLastSelectedPathComponent(), editor, linePositionMap).select();
     }
 
+    /**
+     * Generate Xpath.
+     */
+    private void generateXpath() {
+        if (tree.getLastSelectedPathComponent() instanceof DetailAST) {
+            try {
+                final FileText fileText = new FileText(file, StandardCharsets.UTF_8.name());
+                final DetailAST rootAST = (DetailAST) tree.getModel().getRoot();
+                final DetailAST ast = (DetailAST) tree.getLastSelectedPathComponent();
+                final int defaultTabWidth = 4;
+
+                final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(rootAST,
+                    ast.getLineNo(), ast.getColumnNo(),
+                    fileText, defaultTabWidth);
```

Comment:
it is better to make new ctor
`final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(rootAST);` 
as we need only root AST to do `queryGenerator.generateXpathQuery(ast)`

Reply:
I don't think it will work because, the function I have used is `generateXpathQuery(ast)`. This function uses a number of other methods from the class `XpathQueryGenerator`. It might happen that those method use the other member variables from the existing constructor.

Reply:
`private static String generateXpathQuery(DetailAST ast)` it is static, it means it does not use fields of class, please try.

Reply:
@romani 

Thank you for drawing my attention to this. We don't even need the `rootAST`, since the method is static. We just need the node of which we want to find the xpath.

So, we don't need new constructor.

I have made the method `generateXpathQuery` static again.

Reply:
@romani 

I would also like to ask that I introduced `try catch block` as I was using FileText class constructor which throws an IO exception. Now that it is removed I think I should also remove try catch block. Should I remove it?

Reply:
yes, please remove non required code. The less code the better.

Reply:
Done